File: selenium-webdriver-rspec/spec/01_login_spec.rb
load File.dirname(__FILE__) + '/../test_helper.rb'

describe "User Login" do
  include TestHelper

  before(:all) do
    # for windows, when unable auto-detect firefox binary
    # Please note Firefox on 32 bit is "C:\Program Files (x86)\Mozilla Firefox\firefox.exe"
    # Selenium::WebDriver::Firefox::Binary.path="C:/Program Files/Mozilla Firefox/firefox.exe"
    
    @driver = Selenium::WebDriver.for(browser_type, browser_options)
    driver.manage().window().resize_to(1280, 720)
    driver.manage().window().move_to(30, 78)
    driver.get(site_url)
    
    driver.navigate.to(site_url)
  end

  after(:all) do
    driver.quit unless debugging?
  end

  it "[1] Can sign in OK" do
    goto_page("/login")
    login_page = LoginPage.new(driver)
    login_page.login("agileway", "testwise")  
    # selenium does not have browser.text yet
    try_for(3) {  expect(driver.page_source).to include("Welcome")}
    driver.find_element(:link_text, "Sign off").click
    puts "[stdout] Signed out"
  end

  it "[1] User failed to sign in due to invalid password", :tag => "showcase" do
    goto_page("/login")
    login_page = LoginPage.new(driver)
    login_page.login("agileway", "badpass")
    expect(driver.page_source).to include("Invalid email or password")
  end

end

File: selenium-webdriver-nodejs-mocha/spec/01_login_spec.js
var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;
// var test = require('selenium-webdriver/testing');
var assert = require('assert');

var driver;
const timeOut = 15000;

String.prototype.contains = function(it) {
  return this.indexOf(it) != -1;
};

var helper = require('../test_helper');

// An example in below:  var FlightPage = require('../pages/flight_page.js')
// BEGIN: import pages

// END: import pages

describe('User Authentication', function() {

  before(async function() {
    this.timeout(timeOut);
    driver = new webdriver.Builder().forBrowser('chrome').setChromeOptions(helper.chromeOptions()).build();
    //driver.manage().window().setSize(1280, 720);
    //driver.manage().window().setPosition(30, 78);
  });

  beforeEach(async function() {
    this.timeout(timeOut);
    await driver.get(helper.site_url());
  });

  after(async function() {
    if (!helper.is_debugging()) {
      driver.quit();
    }
  });

  it('[1,2] Invalid user', async function() {
    this.timeout(timeOut);
    await helper.login(driver, "agileway", "badpass")
    await driver.getPageSource().then(function(page_source) {
      assert(page_source.contains("Invalid email or password"))
    });
  });

  it('User can login successfully', async function() {
    this.timeout(timeOut);
    await driver.findElement(webdriver.By.name('username')).sendKeys('agileway');
    await driver.findElement(webdriver.By.name('password')).sendKeys('testwise');
    await driver.findElement(webdriver.By.name('commit')).click();
    await driver.getPageSource().then(function(page_source) {
      assert(page_source.contains("Welcome"))
    });
    await driver.findElement(By.linkText("Sign off")).click();
    
  });


});
